lib/matplotlib/axes/_base.py

```diff
@@ -3902,18 +3902,18 @@ def _make_twin_axes(self, *kl, **kwargs):
 
     def twinx(self):
         """
-        Create a twin Axes sharing the xaxis
+        Create a twin Axes instance sharing the x-axis with self
 
-        Create a twin of Axes for generating a plot with a shared
-        x-axis but independent y-axis.  The y-axis of self will have
-        ticks on left and the returned axes will have ticks on the
-        right. To ensure tick marks of both axis align, see
-        `~matplotlib.ticker.LinearLocator`
+        Create a twin Axes instance for generating a plot with a shared
```

Comment:
An image would be a good idea. These docs feel like they are about implementation rather than the use case - I think what we want to say is "call this method if you want to create a plot with 1 x-axis and 2 y-axes". To me the extra xaxis object is an artifact which is hidden, but apparently there is another use case...